### DISCUSSION TITLE
CPU won't go above 100%

### DISCUSSION THREAD

**Discussion by AurimasNav** [wolveix/satisfactory-server#352] [Q&A] [answered]
I am running satisfactory server on a laptop with 4 cores, however docker desktop never shows container consuming above 100%(sometimes it goes to 102% or 105% but no more). The server is lagging and it is obviously lacking some resources, since memory displays only 1.5GB used out of 12GB I assume the culprit is CPU.

I've configured .wslconfig as well as resource limits and reservations. I can see the memory I've configured reflected in the container stats in docker desktop, but I can't identify why the cpu for container won't go above 100%. Docker desktop displays 400% (4 cpu available).

**Comment by wolveix** [comment, marked as answer]
Hey, this has nothing to do with Docker. As with most game servers, Satisfactory requires strong single-core performance (passmark of 3000+). This is likely your issue.

**Reply by AurimasNav** [reply]
Well that is obvious... but for some reason I assumed that modern software is capable of utilizing more cores :) Other thing that threw me off, windows task manager was showing all 4 cores being utilized at 40-50%, I expected to see one of the cores going 100% when container was displaying 100%, but it never did.

**Reply by wolveix** [reply]
@AurimasNav the devs have explained before that it's due to the nature of the game apparently :) What CPU are you using in your server?

**Reply by AurimasNav** [reply]
It is i5-7200U, before that I've tried  i3-7300, both are lagging, but symptoms are kind of different, the i3 one is heavily rubber-banding, and the i5 does a bit of rubber banding and a lot of unresponsive objects, build actions take few tries, sometimes can't pick up items from conveyors because they do not even highlight as interactive objects.
It is manageable while playing solo, but once we are two, we have to deal with the lags.
From the passmark I see that my i3 has higher single core performance,  but that cpu also does other things..

**Reply by wolveix** [reply]
Yeah, ultimately it looks like a CPU issue unfortunately. While your i3 looks like it _could_ be sufficient, it's a very low-powered CPU overall. If anything else is running on the system (even minor system operations), it'd detract from your game server's performance.

**Comment by AurimasNav** [comment]
For anyone else in similar situation with subpar cpu, I've launched it directly on windows without the docker (steam dedicated server), the performance is night and day so far, at least when playing solo.

**Reply by wolveix** [reply]
@AurimasNav thanks for sharing! Interesting to hear that you're noticing a performance difference. I know Docker for Desktop has quirks, but I didn't think it would impact you that drastically!

**Reply by badnetmask** [reply]
There's a chance your WSL + Docker is the culprit, because I'm running on an i5-8500 with Docker on Linux and it runs just fine.

**Reply by AurimasNav** [reply]
> There's a chance your WSL + Docker is the culprit, because I'm running on an i5-8500 with Docker on Linux and it runs just fine.

I agree the overhead there might be the culprit and it always bothered me that I can't see in windows task manager at least one core hiting 100% when running satisfactory in docker.
